### ISSUE TITLE
(z3str3, ctx-solver-simplify) Segmentation fault (release) or assertion violation at ../src/smt/theory_str_regex.cpp Line: 1386

### ISSUE THREAD

**Issue by muchang** [Z3Prover/z3#4352]
Hi,
For this case, Z3  throws out a segmentation fault:
```
[579] % z3release smt.string_solver=z3str3 small.smt2
Segmentation fault
[580] % z3debug smt.string_solver=z3str3 small.smt2
ASSERTION VIOLATION
File: ../src/smt/theory_str_regex.cpp
Line: 1386
new_lhs
(C)ontinue, (A)bort, (S)top, (T)hrow exception, Invoke (G)DB
a
[581] % 
[581] % cat small.smt2
(declare-fun a () String)
(declare-fun b () String)
(assert (distinct a b ))
(assert (or (distinct (str.len b) 2) (str.in.re a (re.inter (re.+ (str.to.re "a")) (re.comp (re.range "a" "a"))))))
(check-sat-using ctx-solver-simplify)
[582] %
```

OS: Ubuntu 18.04
Commit: fc8dfe3

**Comment by mtrberzi** [comment]
Duplicate of #4344

**Comment by JohnLyu2** [comment]
Fixed in #6000
